src/content/1.7/development/components/grid/_index.md

```diff
@@ -1,324 +1,410 @@
 
-The only method available of `GridDataFactoryInterface` is `getData` which returns an instance of `GridData`.
-A GridData is an **immutable object** used to store and retrieve the grid data, so if you want to alter this data, you must do it in Grid Data Factory *before* the GridData creation.
+// you can access all information thats was defined
+$productsGridDefinition->getColumns(); // collection of defined columns
+$productsGridDefinition->getName(); // "Products"
+$productsGridDefinition->getId(); // "products"
+``` 
 
-You can rely on the implementation provided in the component `GridDataFactory` or create your own if you have specific needs. 
+## Search Criteria
 
-## Grid Factory
+In Grid component Search Criteria is used for Grid's data sorting, paginating & filtering. Search Criteria
+can be loaded from database, URL query or anywhere else. 
 
-Finally the Grid Factory is the final top layer above the other ones, it retrieves the Grid Data thanks to the Grid Data Factory and builds a form to manage filters based on the information in the Grid Definition. Its interface `GridFactoryInterface` implements one method `getGrid`.
+{{% notice info %}}
+Grid component itself does not manage Search Criteria but instead it provides interface for it.
+In PrestaShop [Filters component](#) is used to resolve Search Criteria for Grid.
+{{% /notice %}}
 
-Just like the Grid Data Factory you can either implement your own interface or use the provided `GridFactory` provided in Prestashop core code.
+{{% notice note %}}
+Search Criteria is immutable. This means that once Search Criteria is created it cannot be changed.
+{{% /notice %}}
 
-## Grid services declaration
+### Creating Search Criteria
 
-Only 4 services must be declared: the *Grid Definition Factory*, the *Grid Query Builder*, the *Grid Data Factory*, and the *Grid Factory*.
+Even though most of the time Search Criteria will be created using [Filters component](#), you can still
+create it by yourself. Grid provides simple implementation for it.
 
-```yaml
-# In src/PrestaShopBundle/Resources/config/services/core/grid.yml
-parameters:
-  prestashop.core.grid.data.factory.doctrine_grid_data_factory: PrestaShop\PrestaShop\Core\Grid\Data\Factory\DoctrineGridDataFactory
+```php
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteria;
+
+$filters = [
+    'id_product' => 4,
+    'name' => 'mug',
+];
+
+$searchCriteria = new SearchCriteria(
+    $filters,
+    'id_product',
+    'asc',
+    0,
+    10
+);
+
+$searchCriteria->getFilters();  // $filters array
+$searchCriteria->getOrderBy();  // "id_product"
+$searchCriteria->getOrderWay(); // "asc"
+$searchCriteria->getOffset();   // 0
+$searchCriteria->getLimit();    // 10
+```
 
-services:
-  _defaults:
-    public: true
+When creating Search Criteria you can skip some or all it's data. You can leave default `null` values for order way
+and order by to disable sorting or offset and limit to disable pagination.
```

Comment:
so `offset` has to be nullified to disable pagination? I would have thought nullifying `limit` is more logical

Reply:
if you want to disable pagination, you have to nullify both `limit` and `offset`, you can choose to disable one of them only (e.g. disable offset, but keep limit).

Reply:
I would rephrase this sentence, it's longer but clearer:

When creating Search Criteria you can skip some or all of its data. If you set **both** `orderWay` and `orderBy` to `null` it will disable sorting.  If you set **both** `offset` and `limit` to `null` it will disable pagination.